# Analyzing borrowers’ risk of defaulting

Your project is to prepare a report for a bank’s loan division. You’ll need to find out if a customer’s marital status and number of children has an impact on whether they will default on a loan. The bank already has some data on customers’ credit worthiness.

Your report will be considered when building a **credit scoring** of a potential customer. A ** credit scoring ** is used to evaluate the ability of a potential borrower to repay their loan.

## Open the data file and have a look at the general information. 

In [1]:
import pandas as pd
data = pd.read_csv('/datasets/credit_scoring_eng.csv')


### Conclusion

In [2]:
data.info()
data.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
children            21525 non-null int64
days_employed       19351 non-null float64
dob_years           21525 non-null int64
education           21525 non-null object
education_id        21525 non-null int64
family_status       21525 non-null object
family_status_id    21525 non-null int64
gender              21525 non-null object
income_type         21525 non-null object
debt                21525 non-null int64
total_income        19351 non-null float64
purpose             21525 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


children  days_employed     dob_years  education_id  \
count  21525.000000   19351.000000  21525.000000  21525.000000   
mean       0.538908   63046.497661     43.293380      0.817236   
std        1.381587  140827.311974     12.574584      0.548138   
min       -1.000000  -18388.949901      0.000000      0.000000   
25%        0.000000   -2747.423625     33.000000      1.000000   
50%        0.000000   -1203.369529     42.000000      1.000000   
75%        1.000000    -291.095954     53.000000      1.000000   
max       20.000000  401755.400475     75.000000      4.000000   

       family_status_id          debt   total_income  
count      21525.000000  21525.000000   19351.000000  
mean           0.972544      0.080883   26787.568355  
std            1.420324      0.272661   16475.450632  
min            0.000000      0.000000    3306.762000  
25%            0.000000      0.000000   16488.504500  
50%            0.000000      0.000000   23202.870000  
75%            1.000000      0.000000   32549.611000  
max            4.000000      1.000000  362496.645000

Giving a loan to the customer depends on a lot of things first we are going to ask the the customer according to the data that we have 
for what he need the loan (purpose)?
how many years does he worked or have worked ?
and the total income per month ?
is he married ?
does he have a children ?
is there a debt ?
the rest of the data like gender and education days employed I will take care of the missing values in these columns but not considering them in calculating the credit score 
Therefore first the i read the file.csv
those columns i will be working on ['dob_years','days_employed','total_income','family_status','income_type','children','debt']


## Data preprocessing

### Processing missing values

In [3]:
print(data.isnull().sum())
#quatitive data fill the missing value with mean or median
#data['days_employed'].fillna(value = 0, inplace=True)
null_total_income= data['total_income'].isnull().sum()
null_employed = data['days_employed'].isnull().sum()

lenght = len(data)

rateincome = null_total_income / lenght
ratemploy = null_employed / lenght

print('the percent of the missing value in total income {:.0%}'.format(rateincome)) 
print('the percent of the missing value in days employed {:.0%}'.format(ratemploy)) 

#10% will not affect the data if we delete the rows
#data['total_income'] = data['total_income'].dropna().reset_index(drop=True)
old_mean_income =data['total_income'].mean()
print(old_mean_income)
#data['total_income'] = data['total_income'].fillna(value =old_mean_income)

#new_mean_income =data['total_income'].mean()
    #print(new_mean_income)
print(data.isnull().sum())
if ('type_income' == 'business') | ('type_income' == 'employee') | ('type_income' == 'entrpenuer'):
    data['total_income'] = data['total_income'].fillna(value =old_mean_income)
#else:
 #   data = data.drop()


    

#data['total_income','days_employed'].fillna(value = 0, inplace=True)

#use aggregate function to know how is the missing value affect on the data 
#missing value by none string
#countt = data.loc[data['purpose']=='none','purpose'].count()








children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64
the percent of the missing value in total income 10%
the percent of the missing value in days employed 10%
26787.56835465867
children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64


working with missing value using isnull() and the sum of each null field in the data
according to the result we have 2174 null value in two columns and for me both are important 
days_employed       2174
total_income        2174
calculationg the rate to know if we can drop the rows or not = (10% )
so we can drop them however the total income is linked with the type of income columns so we can deal with the missing values in this column by two methods first we can replace it by the mean (quantitive data) if the customer is employed or business or entreprenuer because ofcourse he has an income while drop the others people unemployed or  student 


In [4]:

countt1 = data.loc[(data['purpose']=='none')|( data['purpose']== ' ')| (data['purpose']=='None'),'purpose'].count()
countt2 = data.loc[(data['education']=='none')|( data['education']== ' ')|(data['education']=='None') ,'education'].count()
countt3 = data.loc[(data['income_type']=='none')|( data['income_type']== ' ')|(data['income_type']=='None') ,'income_type'].count()
print(countt1,countt2,countt3)

0 0 0


make sure that the string columns doesn't have missing value filled by string none or empty space

In [5]:
print(data[data['children']== -1])

print('#of childrens of -1',data[data['children']== -1].count())

data['children']=data['children'].abs()
data[data['children']== 20].replace({20 : 2}, inplace=True)
#mean because maybe he can get a loan the number of children its not that imporant if he has a high income so just put the value zero its not going to affect the data 
#but if i just droped them maybe i will delete rows that we can give them a loan
#no one can have -1 children but if someone have 20 children which is just almost imposible


#data = data.drop(data.loc[(data['children'] == -1) | (data['children'] == 20) ].index)


data['children'].value_counts()

       children  days_employed  dob_years            education  education_id  \
291          -1   -4417.703588         46  secondary education             1   
705          -1    -902.084528         50  secondary education             1   
742          -1   -3174.456205         57  secondary education             1   
800          -1  349987.852217         54  secondary education             1   
941          -1            NaN         57  Secondary Education             1   
1363         -1   -1195.264956         55  SECONDARY EDUCATION             1   
1929         -1   -1461.303336         38  secondary education             1   
2073         -1   -2539.761232         42  secondary education             1   
3814         -1   -3045.290443         26  Secondary Education             1   
4201         -1    -901.101738         41  secondary education             1   
4402         -1  398001.302888         64  SECONDARY EDUCATION             1   
4542         -1   -1811.899756         3

/opt/conda/lib/python3.7/site-packages/pandas/core/frame.py:4263: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  method=method,


0     14149
1      4865
2      2055
3       330
20       76
4        41
5         9
Name: children, dtype: int64

replace the -1 to 1 children and 20 to 2 because its seems like its a human mistake because most of them have a couple
it is look like most of people who have -1 children also have negative days_employed


In [6]:
data['debt'].value_counts()
#values with debt are perfect nothing to do with 

0    19784
1     1741
Name: debt, dtype: int64

In [7]:
#days_employed
#minus days means that he didn't work in them so put them zero it means days he didnt work 

#days employed will not afect the loan credit source so just for modify the value errors



data['years_employed'] = data['days_employed'] / 250.0
#for row in data['years_employed']:
 #   count = 0
  #  sum1 = 0.0
  #  if (data['years_employed'] >0.0) | (data['years_employed'] < 76.0):
     #   sum1 += data['years_employed']
     #   count += 1
    #meann = data['years_employed'].mean()
#def daystoyears(row):
  #  if (row <0.0) | (row>76.0):
   #     row = 0.0
        #row=mean--172year the max yearshould be for age 76 is 76
#data.drop(data.loc[(data['years_employed'] <0.0) | (data['years_employed'] > 76.0)].index)
 #   return row
#data['years_employed'] = data['years_employed'].apply(daystoyears)
data['years_employed'].isnull().sum()

2174

Here we go with the most unrealistic values minus days of work we make this column easier to deal with by making a new column converting it to years 
iin my perspective this column is benifit because each one of employed people will be connected with pesion company while hes wroking so each month he's saving money so even if he retiree now but he has work experince years so ofcourse he has money on 
#so i can calculate the total income with the type of income with the years of employement


In [8]:
data['dob_years'].value_counts()
#we should see if age is below 18 because the min 0 
#bank can't give loan to people under 18 
under18 = data.loc[data['dob_years'] < 18,'dob_years'].count()
print('number of people ages under 18 is',under18)
#we can drop them becasue they are 
print('rate age {:.2%}'. format(under18 /  lenght))
#so we can drop them 
data=data.drop(data.loc[data['dob_years'] < 18].index)
#101 data less than 18 years old so they dont get a loan 


number of people ages under 18 is 101
rate age 0.47%


any customer his age under 18 or have un realistic age should be dropped because age is imporant when givig a loan

### Conclusion

In [9]:
data.describe()


children  days_employed     dob_years  education_id  \
count  21424.000000   19260.000000  21424.000000  21424.000000   
mean       0.542662   63032.838333     43.497479      0.817914   
std        1.375647  140819.014270     12.246934      0.548224   
min        0.000000  -18388.949901     19.000000      0.000000   
25%        0.000000   -2748.301767     33.000000      1.000000   
50%        0.000000   -1203.637301     43.000000      1.000000   
75%        1.000000    -291.172875     53.000000      1.000000   
max       20.000000  401755.400475     75.000000      4.000000   

       family_status_id          debt   total_income  years_employed  
count      21424.000000  21424.000000   19260.000000    19260.000000  
mean           0.971294      0.080891   26794.435857      252.131353  
std            1.419737      0.272673   16493.972601      563.276057  
min            0.000000      0.000000    3306.762000      -73.555800  
25%            0.000000      0.000000   16494.872500      -10.993207  
50%            0.000000      0.000000   23201.873500       -4.814549  
75%            1.000000      0.000000   32539.207750       -1.164691  
max            4.000000      1.000000  362496.645000     1607.021602

first thing to start anlysis the data is to get rid of missing values and errors working on thw whole data 
so to get column by column 
as i enter data.isnull().sum()
children the is no missing values but there was minus children and 20 children means it errors with the data it could be human error or the data engineer but our role is to deal with it so what did i do is ? calculate the number of minus children and the 20 children rest of the data by using value_counts() gives us number from (0-5) which is logical after calculating the percentage of both 20 children plus the minus values gives us less than 10% so we can drop the data but number of children depends with debt both togethe to have a loan or not i mean if i have missing value in the childrn column while all other coulms are corrcet and have a high income so i can give that person a loan and fill the missing vale as 0 i didn't drop them because it doent affect dirrectly on the credit score it i=will affect if the customer have childrn and debt as the same time        
days_employed   there is negative value  and thousands in days so another coulmn that it does not affect the credit score but the experience in work does affect for example if he is a ritree the he has maybe an pension income (the pension income depends on how many years he had worked ) @@@@@@@
dob_years  so any one under age of 18 he can not take a loan logically they are student or unemployed doesnt have a an income and rate of the customers under 18 is less than 10%         
education this column has no missing values however from my perspective have nothing to do with when we are going to deal with bank and loan (we just need to know for whom we are givig the loan so which customer can repay it and here money talks )       
education_id        
family_status the marital satus does affect of having a loan so if the customer didnt pay the spouse pay so the bank almost not losing but we  want to also to check for childrn and debt if he has a children and debt we are going to see the income if its a high income then we can give him a loan but if he has low income then his credit score will be low if he has no debt with medium income he has medium credit score however        
family_status_id    
gender              
income_type  this field is the second imporant column in the data has no missing values but ofcourse there is duplicates but duplicated in this column it doesnt mean that all information are the same so we can not drop without studying the income type column for example if he is a business has no income@@@@@@                    
###########total_income no missing value      
purpose  no missing values ofcourse the are alot of duplicates but that doesnt mean that they have the same values in all fields            
   

### Data type replacement

In [10]:
#data['days_employed'] = pd.to_numeric(data['days_employed'] , downcast = 'int')
data['days_employed'] = data['days_employed'].fillna(0).astype(int)

#data['years_employed'] = pd.to_numeric(data['years_employed'])
data['total_income'] = data['total_income'].fillna(0).astype(int)

#days and income and years
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21424 entries, 0 to 21524
Data columns (total 13 columns):
children            21424 non-null int64
days_employed       21424 non-null int64
dob_years           21424 non-null int64
education           21424 non-null object
education_id        21424 non-null int64
family_status       21424 non-null object
family_status_id    21424 non-null int64
gender              21424 non-null object
income_type         21424 non-null object
debt                21424 non-null int64
total_income        21424 non-null int64
purpose             21424 non-null object
years_employed      19260 non-null float64
dtypes: float64(1), int64(7), object(5)
memory usage: 2.3+ MB


### Conclusion

replace the float values with integeres using as type method 
so i replace the days and years of employent in addition to the roral income

### Processing duplicates

In [11]:
import nltk
import numpy as np
from nltk.stem import SnowballStemmer
english_stemmer = SnowballStemmer('english') 
data['education'] = data['education'].str.lower()
data['purpose'] = data['purpose'].str.lower()
data['income_type'] = data['income_type'].str.lower()
#print(data['purpose'].value_counts())

#commercial =[]real = []real_count = 0rent=[]x=0
english_stemmer = SnowballStemmer('english') 
def purpose_cat(row):
    
    car = ['car']
#car_count=0
    real_estate  =['hous','estat','properti','commerc']
#house_count = 0
    wedding =['wed']
#wed_count=0
    education =['univers','edu']
    for word in row.split(" "):
              
            stemmed_word = english_stemmer.stem(word)
            if stemmed_word in car:
                return 'car'
            elif stemmed_word in real_estate:
                return 'real estate'
            elif stemmed_word in wedding:
                return 'wedding'
            elif stemmed_word in education:
                return 'education'
            


#print(house_count)
#print(wed_count)
print('number of purposes is',data['purpose'].value_counts().count())
print('we are going to categorize them into categorize as follows education, wedding, car ,real estate')
#p= data['purpose'].value_counts().sort_values().reset_index()#.keys().to_list()
#purpose = data['purpose'].value_counts().sort_values().values
#p= data['purpose'].value_counts().sort_values().to_list()
#purpose= npappend(arr=purpose,values=p).to_frame()
#p.columns=['purpose','counts']
data['purpose_s'] = data['purpose'].apply(purpose_cat)
data['purpose_s'].value_counts()



number of purposes is 38
we are going to categorize them into categorize as follows education, wedding, car ,real estate


real estate    10793
car             4293
wedding         2334
education        943
Name: purpose_s, dtype: int64

here i am going to deal with the purpose column by catogorizing it 
however according to the purpose priorty the score will be higher for example if the customer want to buy a house then the repeat will be higher or the bank can invest in it so the bank will benifit according to the purpose then give a higher score according to it.
using the value_counts method on the data['purpose_s'] then save it in a new column 'purpose_s'

### Conclusion

makes every string column lower case 
catogrizing the purpose into four categorize
real estate 
car
wedding 
education
count how much each by count function to see the customer needs the most for a loan
so according to the result most of the people will take a loan for a  real estate and cars
thus the repeit will be much higher therefore will give them more credits

### Categorizing Data

In [12]:
purpose_with_income_type = data.pivot_table(index ='purpose_s',columns='income_type',values='total_income',aggfunc='count')
purpose_with_income_type.fillna(value=0 , inplace=True)
purpose_with_income_type.sort_values(by = 'business',ascending=False,inplace=True)
purpose_with_income_type

income_type  business  civil servant  employee  entrepreneur  \
purpose_s                                                      
real estate    2542.0          754.0    5604.0           1.0   
car            1045.0          285.0    2168.0           0.0   
wedding         527.0          157.0    1217.0           1.0   
education       246.0           49.0     489.0           0.0   

income_type  paternity / maternity leave  retiree  student  unemployed  
purpose_s                                                               
real estate                          0.0   1889.0      1.0         2.0  
car                                  1.0    794.0      0.0         0.0  
wedding                              0.0    432.0      0.0         0.0  
education                            0.0    159.0      0.0         0.0

we can notice that the number of people who have income are retiree employee and civil servant and the business
purpose of business ----> real estate , car , wedding , education
civil servant -----> real estate ,car , wedding , education
employee ----> real estate ,car , wedding , education
retiree -----> real estate ,car , wedding , education
unemployed need to take a real estate but here the credit should be high as unemployed his score is low so unemployed should have poor level priority 


In [13]:
child_debt = data.groupby('debt')['children'].count()
child_debt

debt
0    19691
1     1733
Name: children, dtype: int64

In [14]:
#function to give a income level

    
#function for each level in the score credit     
def income_level(row):
    
    if row > 30000:
        return 'very high'
    elif (row >= 20000) & (row < 30000):
        return 'high'
    elif (row >= 10000) & (row < 20000):
        return 'medium'
    elif (row >= 5000) & (row < 10000):
        return 'low'
    elif row < 5000:
        return 'very low'
   
 #j
data['income_level']  = data['total_income'].apply(income_level)
data['income_level']
#pension year of experinced

0        very high
1           medium
2             high
3        very high
4             high
           ...    
21520    very high
21521         high
21522       medium
21523    very high
21524       medium
Name: income_level, Length: 21424, dtype: object

In [15]:
def scoree(row):
    score = 0
    income = row['income_level']
    typee =  row['income_type']

    if income == 'very high':
            score += 200
            if (typee == 'bussiness') | (type == 'employee'):
                score += 200
            elif typee == 'civil servant':
                score += 150
            elif typee == 'retiree':
                score += 100
            else:
                score += 50
    elif income =='high':
            score += 150
            if (typee == 'bussiness') | (type == 'employee'):
                score += 200
            elif typee == 'civil servant':
                score += 150
            elif typee == 'retiree':
                score += 100
            else:
                score += 50
    elif income =='medium':
            score += 100
            if (typee == 'bussiness') |( type == 'employee'):
                score += 200
            elif typee == 'civil servant':
                score += 150
            elif typee == 'retiree':
                score += 100
            else:
                score += 50
    elif income =='low':
            score += 50
            if (typee == 'bussiness') |( type == 'employee'):
                score += 200
            elif typee == 'civil servant':
                score += 150
            elif typee == 'retiree':
                score += 100
            else:
                score += 50
    elif income =='very low':
            score += 10
            if( typee == 'bussiness' )| (type == 'employee'):
                score += 200
            elif typee == 'civil servant':
                score += 150
            elif typee == 'retiree':
                score += 100
            else:
                score += 50
 
    return score 
data['score'] = data.apply(scoree,axis=1)
data['score']

0        250
1        150
2        200
3        250
4        250
        ... 
21520    250
21521    250
21522    150
21523    250
21524    150
Name: score, Length: 21424, dtype: int64

In [16]:
def level(row):
    if row >= 600: 
        return 'excellent'
    elif (row >= 450) & (row < 600):
        return 'very good'
    elif row >= 300 & row < 450:
        return 'good'

    elif row >= 150 & row < 300:
        return 'fair'
    return 'poor'
data['priority_level'] = data['score'].apply(level)
data['priority_level']
        

0        good
1        good
2        good
3        good
4        good
         ... 
21520    good
21521    good
21522    good
21523    good
21524    good
Name: priority_level, Length: 21424, dtype: object

calculating the score according to income type , total income , years of experience
high priority take score of 200 each 
good takes 150
poor 100
fair 50 for each condition bu summing up how much each customer   sums score then we
apply the score credit to the level credit to know whos credit is exellent to fair

### Conclusion

by categorizing the data we can notice to whom we should give more credit score four types of income_type have a total income the business and the employee civil servant and retiree according to whom have more income we will give him more score thus he's priorty level is high for taking a loan

## Answer these questions

- Is there a relation between having kids and repaying a loan on time?

In [17]:
#pivot table of purpose and income_type so if we gather between two information we are going to know if the person work then 
#he can repay the loan 
#the second pivot _table to see if he there is a relationship between debt and being married or having a children
debt_family_status = data.pivot_table(index ='family_status',columns=['children'],values='debt',aggfunc=['sum','count'])
debt_family_status_ = data.pivot_table(index ='family_status',columns=['debt'],values='children',aggfunc=['sum','count'])
debt_family_status_children = data.pivot_table(index ='children',values='debt',aggfunc=['sum','count'])

debt_family_status.fillna(value=0 , inplace=True)
print(debt_family_status)


                     sum                                      count          \
children              0      1      2     3    4    5    20      0       1    
family_status                                                                 
civil partnership  227.0  118.0   30.0   8.0  0.0  0.0  3.0  2739.0   997.0   
divorced            55.0   21.0    7.0   1.0  0.0  0.0  1.0   777.0   315.0   
married            514.0  245.0  145.0  17.0  3.0  0.0  3.0  7469.0  3004.0   
unmarried          210.0   51.0    9.0   1.0  1.0  0.0  1.0  2252.0   452.0   
widow / widower     52.0    7.0    3.0   0.0  0.0  0.0  0.0   843.0    81.0   

                                                   
children               2      3     4    5     20  
family_status                                      
civil partnership   342.0   56.0   8.0  2.0  12.0  
divorced             79.0   11.0   1.0  0.0   2.0  
married            1527.0  247.0  29.0  7.0  48.0  
unmarried            74.0    8.0   2.0  0.0   9.0  
wid

In [18]:
child_debt = data.groupby('debt')['children'].count()
child_debt
rate_no_debts = (child_debt/len(child_debt))/100
rate_no_debts
child_debt_s = data.groupby('income_type')['children'].sum()
child_debt_s

income_type
business                       3050
civil servant                   911
employee                       7137
entrepreneur                      0
paternity / maternity leave       2
retiree                         525
student                           0
unemployed                        1
Name: children, dtype: int64

### Conclusion

children do not factor into credit score calculations. Instead, it's the extra money you spend from your income i mean if you dont have children you can save more money,however having a children requires you to  debt as we can see that people with no debt do have 98% children
however people with debts have 8% of them children
however employee and business people have children and have total income so even if they have children they still can repay the payments on time 

- Is there a relation between marital status and repaying a loan on time?

In [19]:
marital = data.pivot_table(columns ='family_status',index='score',values='family_status_id',aggfunc={'count'})
print(marital)
df =data.groupby('family_status')['score'].count()
print(df)

                          count                                           
family_status civil partnership divorced married unmarried widow / widower
score                                                                     
60                          339       86     929       224              37
100                          84       20     303        44              15
110                          87       20     226        41              53
150                         985      270    2773       714             170
160                          20        5      95        24               4
200                        1202      342    3528       826             323
250                        1166      352    3606       765             248
300                         197       63     617       107              95
350                          76       27     254        52              10
family_status
civil partnership     4156
divorced              1185
married              12331
unmar

### Conclusion

no it doesnt affect unless people who are married and have joint accounts will find that their partner's situation will be considered even if they are not applying for a joint loan
Here most of people who have very high to medium income are married or civil partnership
therefore there credits will be high so they have a high priorty of giving the loan too

- Is there a relation between income level and repaying a loan on time?

In [20]:
leve = data.pivot_table(index='income_level',values='score',aggfunc={'count'})
leve

count
income_level       
high           6038
low             895
medium         6411
very high      5890
very low       2190

### Conclusion

credit score is depends the most on the income more income more credit 
so people with very high income level are 5890 
high level = 6038 
medium level are 6411 
so most of people have medium level credit score 
second thing it depends on the income type as the customer work then always there is an income means there credit will be high 
income type have evidence of stability – for example, you’re employed or business 
the score depends on the total income an type of income

- How do different loan purposes affect on-time repayment of the loan?

In [21]:
purpose_with_income_type


income_type  business  civil servant  employee  entrepreneur  \
purpose_s                                                      
real estate    2542.0          754.0    5604.0           1.0   
car            1045.0          285.0    2168.0           0.0   
wedding         527.0          157.0    1217.0           1.0   
education       246.0           49.0     489.0           0.0   

income_type  paternity / maternity leave  retiree  student  unemployed  
purpose_s                                                               
real estate                          0.0   1889.0      1.0         2.0  
car                                  1.0    794.0      0.0         0.0  
wedding                              0.0    432.0      0.0         0.0  
education                            0.0    159.0      0.0         0.0

### Conclusion

according to a specific purpose the bank can give time for repayment or it can investigate in the purpose like( buy the customer need and sell to him )
so the purposes we  have is 
real state it can investigate  in the property or if was a commercial business the bank can give a period of time to make the commercial success then take a high repeat on every payemnt
buy a car : if the customer needs to buy a car the bank can buy this car and sell it back in a higher price plus high repeat 
wedding in this purpose the bank can make the repeat high because wedding are important and its very imporatnt for people so they will take the loan with agreement of high repeat but here it will depends on the type income and the total income of the customer if it was high or good enough to repay it on time 
education repayemnt 


## General conclusion

In this report you can find out which customer can take a loan or the bank could benefit from him by calculating his score and priorty level of taking a loan 

when we are dealing with money every percent in data is important i read the data knowing which coulumns have null or missing value trying to figure out the reason behind each problem on the other hand trying to solve it

when in comes to deal with the data we have several points we should take in perspective

 first investigating the missing value instead of dropping them knowing which is human error like the days-employed and the number of children
 
 then start dealing with the data categorizing tables to make the analysis much easier to anlyze so according to the anlysis we have found that credit score depends on the first priority on the income any person who have stable income with no debts looking in the marital status and the family satus could have a loan 
 according to the purpose the bank can know which the greatest purpose the people come to take loan in according to th data it was the real estate so the bank should think in a way that to benefit from the purpoe like investing in them and increase the credit score 
 
 children and marital status depends in having a children how many and the most important if they do have a debts
 
 we found that the days of employement with negative value are the same with people who have negative children